doc/Language/typesystem.pod6

```diff
@@ -258,6 +258,49 @@ you must use private attribute notation instead.
     C.new.say; C.new('answer').say;
     # OUTPUT: «C.new(attr => 42)␤
     #          C.new(attr => "answer")␤»
+    
+=head4 submethod TWEAK
+
+The L<submethod|/type/Submethod> C<TWEAK> is (indirectly) called by
+L<.bless|/type/Mu#method_bless>. It
+is meant to set private and public attributes of a class after the BUILD submethod has been called. The default
+constructor L<.new|/type/Mu#method_new> defined in C<Mu> is the method that
+invokes it. Given that public accessor methods are not available in C<TWEAK>,
+you must use private attribute notation instead.
+
+The difference between BUILD and TWEAK is that not all the attributes in an object have been built.
+In addition, when there is inheritance the BUILD submethods of the child classes are called after
```

Comment:
Not really true either. You can build as many as you want in BUILD, or bind them in the call. But `self` is not available, so computed attributes might need to be defined later, during TWEAK.

Reply:
JJ: I have run into the problem quite a lot. I am creating a class. I BUILD an object (not the phaser). But the attribute has a default that has not been applied in BUILD. But it is there in TWEAK. That's what my example shows.

Reply:
You can apply the default in the BUILD signature, but the point is that since self has not been built, no defaults are available. It's not the values, it's the object. The whole point of TWEAK is that the object is already built. Same thing when an attribute is required: you need to assign a value during BUILD; TWEAK is too late.

Reply:
Yes it is possible to put the default in the BUILD signature, and also in the has declaration. 

The whole point is that BUILD and TWEAK offer different flexibililties. I know I could not do some things with BUILD alone. And there is no documentation for TWEAK.

Reply:
Let's please stick to this sentence.

> The difference between BUILD and TWEAK is that not all the attributes in an object have been built.

First "the difference... is that not all the attributes" should be preceded by a subject that said *where* they have not been built. But even so, it's not true. All attributes can be built after BUILD. It's not named BUILD for no reason. And TWEAK is named TWEAK precisely for that reason: can be used for after-build tweaks, checks, whatever. The emphasis in in after-**build**: the object has already been built. You can tweak it afterwards, but what TWEAK receives is a perfectly fine object.